### PR TITLE
Latex index

### PR DESCRIPTION
**Pull request by AbcAeffchen**
Added an index to all hashtags.
I also added some postproc filters to fix some utf8 quotation marks the latex file cannot be compiled with.

### CODE REVIEW

File: rednotebook/util/markup.py [hunk 1]
```diff
@@ -265,6 +269,17 @@ def _get_config(target, options):
         config['preproc'].append([r'\\\(\s*(.+?)\s*\\\)', r"BEGINMATH''\1''ENDMATH"])
         config['postproc'].append([r'BEGINMATH(.+)ENDMATH', r'$\1$'])
 
+        # Fix utf8 quotations - „, “ and ” cause problems compiling the latex document
+        config['postproc'].append([u'„', '"'])
+        config['postproc'].append([u'”', '"'])
+        config['postproc'].append([u'“', '"'])
```
Comment (jendrikseipp) [inline rednotebook/util/markup.py]: I don't think it's worth adding special cases for these symbols. There are too many symbols that Latex will stumble over and we can't list them all here.

File: rednotebook/util/markup.py [hunk 2]
```diff
@@ -265,6 +269,17 @@ def _get_config(target, options):
...
+
+        # enabling index
```
Comment (jendrikseipp) [inline rednotebook/util/markup.py]: Comments should be in imperative mode. If they are a sentence, they should end with a full stop, so in this case `# Enable index.`. (Commit messages should also use imperative mode BTW.)

File: rednotebook/util/markup.py [hunk 3]
```diff
@@ -265,6 +269,17 @@ def _get_config(target, options):
...
+        config['style'].append('makeidx')
+        config['postproc'].append([r'BEGININDEX(.+?)ENDINDEX', r'\\index{\1}'])
+        config['postproc'].append(['begin{document}', 'makeindex\n\\\\begin{document}'])
+        config['postproc'].append(['end{document}', 'printindex\n\n\\\\end{document}'])
```
Comment (jendrikseipp) [inline rednotebook/util/markup.py]: Can you please add documentation on how to produce a PDF with an index to the "Convert Latex output to PDF" section in `rednotebook/info.py`?

File: rednotebook/util/markup.py (hunk 1 above)
Comment (AbcAeffchen) [inline rednotebook/util/markup.py]: That is true. But if you copy a lot from the web, e.g. news articles, you get a ton of this three symbols. If you also change the encoding to `utf8x`, there are only a few symbols left to cause problems.

Comment (jendrikseipp) [inline rednotebook/util/markup.py]: OK, let's leave this in then. Can you turn it into a single `config['postproc'].extend(...)` call, please?

Comment (AbcAeffchen) [inline rednotebook/util/markup.py]: Sure. Should I also add the change from `utf8` to `utf8x`?

**Comment by jendrikseipp** [comment]
Please keep utf8 instead if utf8x. utf8x seems to cause all kinds of 
problems 
(https://tex.stackexchange.com/questions/13067/utf8x-vs-utf8-inputenc).

**Comment by AbcAeffchen** [comment]
I'm sorry that it took so much time. I was very busy. And somehow I screwed up a rebase I think...

**Comment by AbcAeffchen** [comment]
Lets close this for now, I open a new one after I fixed it

**Pull request closed without merging**